primitives/trie/src/storage_proof.rs

```diff
@@ -54,10 +58,16 @@ impl StorageProof {
 		self.trie_nodes.is_empty()
 	}
 
+	/// Convert into an iterator over encoded trie nodes in lexicographical order constructed
+	/// from the proof.
+	pub fn iter_nodes(self) -> IntoIter<Vec<u8>> {
+		self.trie_nodes.into_iter()
+	}
+
 	/// Create an iterator over encoded trie nodes in lexicographical order constructed
 	/// from the proof.
-	pub fn iter_nodes(self) -> StorageProofNodeIterator {
-		StorageProofNodeIterator::new(self)
+	pub fn iter(&self) -> Iter<'_, Vec<u8>> {
```

Comment:
Could be rename `iter_nodes`

Reply:
Yeah, this rename doesn't make any sense.

Reply:
And here return `impl Iter`

Reply:
> And here return `impl Iter`

Yeah, But I think user maybe wanna use more trait methods here.

Reply:
Sounds good.

Reply:
We didn't supported this before. So, not sure we need this now.